server/webapp/WEB-INF/rails.new/webpack/views/preferences/add_filter.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+;(function () { // eslint-disable-line no-extra-semi
+  "use strict";
+
+  const m = require("mithril");
+  const _ = require("lodash");
+
+  function AddFilter(filterModels, pipelineModel) {
+    return {
```

Comment:
Can we keep it as a simple mithril component, and pass the arguments to the component
and use them via `node.attrs`.

So, in views, we can simply mount the `AddFilter` component with `filterModels` and `pipelineModel` as arguments.

[here](https://github.com/gocd/gocd/pull/3360/files#diff-a9a0b22b9e90646fe6f9333fc052c899R77) it could be `<AddFilterView filterModels={filterModels} pipelineModel={pipelineModel} />`